src/VirtualClient/VirtualClient.Main/ExecuteProfileCommand.cs

```diff
@@ -825,9 +826,113 @@ private void InitializeProfile(ExecutionProfile profile)
 
             ValidationResult result = ExecutionProfileValidation.Instance.Validate(profile);
             result.ThrowIfInvalid();
+
+            // Process conditional parameters (ParametersOn feature)
+            await this.ProcessParametersOnAsync(profile, dependencies);
```

Comment:
Similarly to how there is an extension method ExecutionProfile.Inline(), the logic to determine which parameters should be used from the conditional set should be an extension method as well. You may want to just extend the Inline() method to take in the IServiceCollection.

Recommendation:
I know that there are dependencies in IServiceCollection that cannot be referenced from the VirtualClient.Contracts project. We can likely just move the extensions method class to the VirtualClient.Core project to address the scenario.